### PR TITLE
[curl] Fix #14049

### LINKED ISSUE #14049: curl:x64-linux-dynamic build failure
**Host Environment**
- Ubuntu 20.04.1 LTS
- Compiler: clang version 10.0.0-4ubuntu1

**To Reproduce**
`./vcpkg install curl:x64-linux-dynamic`

with the triplet `x64-linux-dynamic` described in your tutorial, i.e.

```cmake
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)

set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```

```bash
-- Performing post-build validation
There should be no empty directories in /home/klalumiere/Projects/vcpkg/packages/curl_x64-linux-dynamic
The following empty directories were found:

    /home/klalumiere/Projects/vcpkg/packages/curl_x64-linux-dynamic/debug/bin
    /home/klalumiere/Projects/vcpkg/packages/curl_x64-linux-dynamic/bin

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/dir" "${CURRENT_PACKAGES_DIR}/some/other/dir")


Found 1 error(s). Please correct the portfile:
    /home/klalumiere/Projects/vcpkg/ports/curl/portfile.cmake
-- Performing post-build validation done
Error: Building package curl:x64-linux-dynamic failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: curl:x64-linux-dynamic
  Vcpkg version: 2020.06.15-unknownhash

Additionally, attach any relevant sections from the log files above.
```

vcpkg SHA: `790910f79f653978f90aadd958abf3c407215552`

I'm going to solve the issue and submit a pull request as soon as possible.

### PR DESCRIPTION
**Pull request by klalumiere**
- What does your PR fix? Fixes #14049

- Which triplets are supported/not supported? Adds support for the non-official `x64-linux-dynamic` which is described in the official doc.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -86,7 +86,7 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/CURL)
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR ${VCPKG_CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```
Comment (PhoebeHui) [inline ports/curl/portfile.cmake:89]:
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR VCPKG_TARGET_IS_LINUX)
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Could you also update the Port-Verison in CONTROL file?

From
Version: 7.73.0

To 
Version: 7.73.0
Port-Version: 1

**Comment by klalumiere** [comment]
@PhoebeHui Sorry for the `requested review`, I did not notice you already approved 😳.

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
@klalumiere, never mind, appreciate your effort very much.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**